### ISSUE TITLE
ebpf_assert(!_ebpf_native_client_table || ebpf_hash_table_key_count(_ebpf_native_client_table) == 0)

### ISSUE THREAD

**Issue by Alan-Jowett** [microsoft/ebpf-for-windows#1555]
```
 # Child-SP          RetAddr               Call Site
00 000000ef`50f8e248 00007ffd`e6f58db3     ntdll!ZwWaitForMultipleObjects+0x14 [minkernel\ntdll\daytona\objfre\amd64\usrstubs.asm @ 947] 
01 000000ef`50f8e250 00007ffd`e6f58cce     KERNELBASE!WaitForMultipleObjectsEx+0xd3 [minkernel\kernelbase\synch.c @ 1551] 
02 000000ef`50f8e530 00007ffd`e87019ef     KERNELBASE!WaitForMultipleObjects+0xe [minkernel\kernelbase\synch.c @ 1403] 
03 000000ef`50f8e570 00007ffd`e870142e     kernel32!WerpReportFaultInternal+0x587 [onecore\windows\feedback\core\faultrep\lib\faultrep.cpp @ 1073] 
04 000000ef`50f8e690 00007ffd`e704fbbb     kernel32!WerpReportFault+0xbe [onecore\windows\feedback\core\faultrep\lib\faultrep.cpp @ 1384] 
05 000000ef`50f8e6d0 00007ffd`e9b1bdb7     KERNELBASE!UnhandledExceptionFilter+0x3db [minkernel\kernelbase\xcpt.c @ 701] 
06 (Inline Function) --------`--------     ntdll!RtlpThreadExceptionFilter+0x9b [minkernel\ntdll\rtlstrt.c @ 960] 
07 000000ef`50f8e7f0 00007ffd`e9b01e57     ntdll!RtlUserThreadStart$filt$0+0xac [minkernel\ntdll\rtlstrt.c @ 1193] 
08 000000ef`50f8e830 00007ffd`e9b1716f     ntdll!__C_specific_handler+0x97 [minkernel\crts\crtw32\misc\riscchandler.c @ 433] 
09 000000ef`50f8e8a0 00007ffd`e9aad0af     ntdll!RtlpExecuteHandlerForException+0xf [minkernel\ntos\rtl\amd64\xcptmisc.asm @ 132] 
0a 000000ef`50f8e8d0 00007ffd`e9acfca5     ntdll!RtlDispatchException+0x29f [minkernel\ntos\rtl\amd64\exdsptch.c @ 669] 
0b 000000ef`50f8f030 00007ffd`e6f2d4fc     ntdll!RtlRaiseException+0x195 [minkernel\ntos\rtl\amd64\raise.c @ 240] 
*** WARNING: Unable to verify checksum for unit_tests.exe
0c 000000ef`50f8fed0 00007ff7`ebcd7035     KERNELBASE!RaiseException+0x6c [minkernel\kernelbase\xcpt.c @ 954] 
0d 000000ef`50f8ffb0 00007ffd`42636b51     unit_tests!_wer_report_test::_terminate_hook+0x25 [E:\ebpf-for-windows\tests\libs\util\wer_report.hpp @ 77] 
0e 000000ef`50f8ffe0 00007ffd`42635330     ucrtbased!_VCrtDbgReportW+0x841 [minkernel\crts\ucrt\src\appcrt\misc\dbgrptt.cpp @ 613] 
0f 000000ef`50f98120 00007ffd`42622e51     ucrtbased!_CrtDbgReportW+0x60 [minkernel\crts\ucrt\src\appcrt\misc\dbgrpt.cpp @ 278] 
10 000000ef`50f98180 00007ffd`42623003     ucrtbased!issue_debug_notification+0x31 [minkernel\crts\ucrt\src\appcrt\internal\report_runtime_error.cpp @ 25] 
11 000000ef`50f981d0 00007ffd`4263ab0d     ucrtbased!__acrt_report_runtime_error+0x13 [minkernel\crts\ucrt\src\appcrt\internal\report_runtime_error.cpp @ 154] 
12 000000ef`50f98230 00007ffd`4264094f     ucrtbased!abort+0x1d [minkernel\crts\ucrt\src\appcrt\startup\abort.cpp @ 61] 
13 000000ef`50f98270 00007ffd`4263e881     ucrtbased!common_assert_to_stderr<wchar_t>+0xbf [minkernel\crts\ucrt\src\appcrt\startup\assert.cpp @ 187] 
14 000000ef`50f982d0 00007ffd`4264158f     ucrtbased!common_assert<wchar_t>+0x41 [minkernel\crts\ucrt\src\appcrt\startup\assert.cpp @ 415] 
15 000000ef`50f98310 00007ff7`ebdf4ea0     ucrtbased!_wassert+0x2f [minkernel\crts\ucrt\src\appcrt\startup\assert.cpp @ 444] 
16 000000ef`50f98340 00007ff7`ebddc267     unit_tests!ebpf_native_terminate+0x110 [E:\ebpf-for-windows\libs\execution_context\ebpf_native.c @ 300] 
17 000000ef`50f98440 00007ff7`ebc6a4dc     unit_tests!ebpf_core_terminate+0x57 [E:\ebpf-for-windows\libs\execution_context\ebpf_core.c @ 215] 
18 000000ef`50f98470 00007ff7`ebfe2eec     unit_tests!_test_helper_end_to_end::~_test_helper_end_to_end+0x41c [E:\ebpf-for-windows\tests\end_to_end\test_helper.cpp @ 560] 
19 000000ef`50f987b0 00007ffd`dba81030     unit_tests!`bindmonitor_tailcall_test'::`1'::dtor$0+0x1c
1a 000000ef`50f987f0 00007ffd`dba841d7     VCRUNTIME140_1D!_CallSettingFrame+0x20 [d:\a01\_work\12\s\src\vctools\crt\vcruntime\src\eh\amd64\handlers.asm @ 50] 
1b 000000ef`50f98820 00007ffd`dba8659b     VCRUNTIME140_1D!__FrameHandler4::FrameUnwindToState+0x2b7 [d:\a01\_work\12\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 1104] 
1c 000000ef`50f98930 00007ffd`dba82cd2     VCRUNTIME140_1D!__FrameHandler4::FrameUnwindToEmptyState+0x4b [d:\a01\_work\12\s\src\vctools\crt\vcruntime\src\eh\risctrnsctrl.cpp @ 234] 
1d 000000ef`50f98970 00007ffd`dba86ceb     VCRUNTIME140_1D!__InternalCxxFrameHandler<__FrameHandler4>+0x282 [d:\a01\_work\12\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 312] 
1e 000000ef`50f98a40 00007ff7`ebe55830     VCRUNTIME140_1D!__CxxFrameHandler4+0xfb [d:\a01\_work\12\s\src\vctools\crt\vcruntime\src\eh\risctrnsctrl.cpp @ 304] 
1f 000000ef`50f98ad0 00007ffd`e9b171ef     unit_tests!__GSHandlerCheck_EH4+0x90 [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\gs\amd64\gshandlereh4.cpp @ 73] 
20 000000ef`50f98b20 00007ffd`e9aae3f3     ntdll!RtlpExecuteHandlerForUnwind+0xf [minkernel\ntos\rtl\amd64\xcptmisc.asm @ 254] 
21 000000ef`50f98b50 00007ffd`dba8696f     ntdll!RtlUnwindEx+0x2f3 [minkernel\ntos\rtl\amd64\exdsptch.c @ 1537] 
22 000000ef`50f99280 00007ffd`dba81c1e     VCRUNTIME140_1D!__FrameHandler4::UnwindNestedFrames+0x1af [d:\a01\_work\12\s\src\vctools\crt\vcruntime\src\eh\risctrnsctrl.cpp @ 684] 
23 000000ef`50f99380 00007ffd`dba8218b     VCRUNTIME140_1D!CatchIt<__FrameHandler4>+0x12e [d:\a01\_work\12\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 1328] 
24 000000ef`50f99420 00007ffd`dba82ec5     VCRUNTIME140_1D!FindHandler<__FrameHandler4>+0x55b [d:\a01\_work\12\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 637] 
25 000000ef`50f99600 00007ffd`dba86ceb     VCRUNTIME140_1D!__InternalCxxFrameHandler<__FrameHandler4>+0x475 [d:\a01\_work\12\s\src\vctools\crt\vcruntime\src\eh\frame.cpp @ 355] 
26 000000ef`50f996d0 00007ff7`ebe55830     VCRUNTIME140_1D!__CxxFrameHandler4+0xfb [d:\a01\_work\12\s\src\vctools\crt\vcruntime\src\eh\risctrnsctrl.cpp @ 304] 
27 000000ef`50f99760 00007ffd`e9b1716f     unit_tests!__GSHandlerCheck_EH4+0x90 [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\gs\amd64\gshandlereh4.cpp @ 73] 
28 000000ef`50f997b0 00007ffd`e9aad0af     ntdll!RtlpExecuteHandlerForException+0xf [minkernel\ntos\rtl\amd64\xcptmisc.asm @ 132] 
29 000000ef`50f997e0 00007ffd`e9acfca5     ntdll!RtlDispatchException+0x29f [minkernel\ntos\rtl\amd64\exdsptch.c @ 669] 
2a 000000ef`50f99f40 00007ffd`e6f2d4fc     ntdll!RtlRaiseException+0x195 [minkernel\ntos\rtl\amd64\raise.c @ 240] 
2b 000000ef`50f9ade0 00007ffd`c968b460     KERNELBASE!RaiseException+0x6c [minkernel\kernelbase\xcpt.c @ 954] 
2c 000000ef`50f9aec0 00007ff7`ebcdfd0a     VCRUNTIME140D!_CxxThrowException+0x120 [d:\a01\_work\12\s\src\vctools\crt\vcruntime\src\eh\throw.cpp @ 75] 
2d 000000ef`50f9af50 00007ff7`ebbf2134     unit_tests!Catch::AssertionHandler::complete+0x5a [E:\ebpf-for-windows\external\Catch2\src\catch2\internal\catch_assertion_handler.cpp @ 57] 
2e 000000ef`50f9af90 00007ff7`ebc1d109     unit_tests!_single_instance_hook::_single_instance_hook+0x384 [E:\ebpf-for-windows\tests\end_to_end\helpers.h @ 99] 
2f 000000ef`50f9b190 00007ff7`ebbfe150     unit_tests!bindmonitor_tailcall_test+0x1c79 [E:\ebpf-for-windows\tests\end_to_end\end_to_end.cpp @ 720] 
30 000000ef`50f9e690 00007ff7`ebcf8ad2     unit_tests!CATCH2_INTERNAL_TEST_21+0x10 [E:\ebpf-for-windows\tests\end_to_end\end_to_end.cpp @ 903] 
31 000000ef`50f9e6c0 00007ff7`ebcf0e11     unit_tests!Catch::TestInvokerAsFunction::invoke+0x12 [E:\ebpf-for-windows\external\Catch2\src\catch2\internal\catch_test_case_registry_impl.cpp @ 150] 
32 000000ef`50f9e6f0 00007ff7`ebce9f67     unit_tests!Catch::TestCaseHandle::invoke+0x21 [E:\ebpf-for-windows\external\Catch2\src\catch2\catch_test_case_info.hpp @ 115] 
33 000000ef`50f9e720 00007ff7`ebce9d90     unit_tests!Catch::RunContext::invokeActiveTestCase+0x47 [E:\ebpf-for-windows\external\Catch2\src\catch2\internal\catch_run_context.cpp @ 508] 
34 000000ef`50f9e770 00007ff7`ebce81dc     unit_tests!Catch::RunContext::runCurrentTest+0x250 [E:\ebpf-for-windows\external\Catch2\src\catch2\internal\catch_run_context.cpp @ 475] 
35 000000ef`50f9eac0 00007ff7`ebf990d7     unit_tests!Catch::RunContext::runTest+0x2bc [E:\ebpf-for-windows\external\Catch2\src\catch2\internal\catch_run_context.cpp @ 239] 
36 000000ef`50f9ef30 00007ff7`ebf98349     unit_tests!Catch::`anonymous namespace'::TestGroup::execute+0xe7 [E:\ebpf-for-windows\external\Catch2\src\catch2\catch_session.cpp @ 110] 
37 000000ef`50f9f090 00007ff7`ebf97de0     unit_tests!Catch::Session::runInternal+0x409 [E:\ebpf-for-windows\external\Catch2\src\catch2\catch_session.cpp @ 335] 
38 000000ef`50f9f4f0 00007ff7`ebe58df2     unit_tests!Catch::Session::run+0x50 [E:\ebpf-for-windows\external\Catch2\src\catch2\catch_session.cpp @ 263] 
39 000000ef`50f9f530 00007ff7`ebe58d61     unit_tests!Catch::Session::run<char>+0x52 [E:\ebpf-for-windows\external\Catch2\src\catch2\catch_session.hpp @ 41] 
3a 000000ef`50f9f570 00007ff7`ebe55d69     unit_tests!main+0x61 [E:\ebpf-for-windows\external\Catch2\src\catch2\internal\catch_main.cpp @ 36] 
3b 000000ef`50f9f790 00007ff7`ebe55cbe     unit_tests!invoke_main+0x39 [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl @ 79] 
3c 000000ef`50f9f7e0 00007ff7`ebe55b7e     unit_tests!__scrt_common_main_seh+0x12e [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl @ 288] 
3d 000000ef`50f9f850 00007ff7`ebe55dde     unit_tests!__scrt_common_main+0xe [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl @ 331] 
3e 000000ef`50f9f880 00007ffd`e86a335d     unit_tests!mainCRTStartup+0xe [D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_main.cpp @ 17] 
3f 000000ef`50f9f8b0 00007ffd`e9ad7f58     kernel32!BaseThreadInitThunk+0x1d [clientcore\base\win32\client\thread.c @ 75] 
40 000000ef`50f9f8e0 00000000`00000000     ntdll!RtlUserThreadStart+0x28 [minkernel\ntdll\rtlstrt.c @ 1192] 
```

**Comment by Alan-Jowett** [comment]
Might be a duplicate of #1509

**Comment by Alan-Jowett** [comment]
Blocked on #1509
